### ISSUE TITLE
The lib restricts links on resource level to 'self' only

### ISSUE THREAD

**Issue by neomerx** [neomerx/json-api#64]
spotted by @yoanisgil  in #54 

Not very often used feature however it should give options to add custom links on resource level

``` json
{
  "data": 
  {
    "type": "someType",
    "id": "123",
    "attributes": {
      "name": "some name"
    },
    "links": {
        "custom-link": "..." 
    }
  }
}
```

Links on 'relationships' level could be used while it's not added

**Comment by yoanisgil** [comment]
@neomerx I think the example should rather read:

``` json
{
  "data": [
    {
      "type": "someType",
      "id": "123",
      "attributes": {
        "name": "some name"
      },
      "links": {
        "custom-link": "..."
      }
    }
  ]
}
```

The example you provided is already possible with $encoder->withLinks([ ...])

**Comment by neomerx** [comment]
It's possbile to have single resource in 'data' section and 'links' section is on the same level as 'attributes' one. I think both examples are correct.

**Comment by yoanisgil** [comment]
Yes you're right. Let me know when you have something that works and I will give it a try ;)

**Comment by neomerx** [comment]
@yoanisgil In `Schema` you can override the following two methods to add/replace default resource links. The code in `develop` branch

``` php
    /**
     * Get links related to resource.
     *
     * @param mixed $resource
     *
     * @return array Array key is link name and value is LinkInterface.
     */
    public function getResourceLinks($resource);

    /**
     * Get links related to resource when it is in 'included' section.
     *
     * @param mixed $resource
     *
     * @return array Array key is link name and value is LinkInterface.
     */
    public function getIncludedResourceLinks($resource);
```

In your case it might be something like

``` php

public function getResourceLinks($resource)
{
    return [
        // use $resource to compose the sub-url
        'configuration' => new Link('/domains/...'),
    ];
}

```

**UPD** By default `getResourceLinks` returns array with `self` link and `getIncludedResourceLinks` returns empty array of links.

**Comment by yoanisgil** [comment]
Ok. Let me wrap around a couple of fixes and I will give it a try. Thanks for the quick feedback!

**Comment by yoanisgil** [comment]
@neomerx It works like a charm! Let me know once you merge into master and tag so I can update my composer.json (right now is pointing to dev-develop)

**Comment by neomerx** [comment]
Doc updated https://github.com/neomerx/json-api/wiki/Schemas#show-links-related-to-resource

**Comment by neomerx** [comment]
@yoanisgil Released in 0.5.12

**Comment by yoanisgil** [comment]
Thanks again for taking the time to address this issue.
